multidb/middleware.py

```diff
@@ -13,7 +13,7 @@
 PINNING_SECONDS = int(getattr(settings, 'MULTIDB_PINNING_SECONDS', 15))
 
 
-READ_ONLY_METHODS = ('GET', 'TRACE', 'HEAD', 'OPTIONS')
+READ_ONLY_METHODS = frozenset([GET', 'TRACE', 'HEAD', 'OPTIONS'])
```

Comment:
```python
frozenset([GET
           ^
```

Just missing a quote, that explains all the test failures. Mind fixing and squashing it into one commit?